runtime/src/bank.rs

```diff
@@ -16031,6 +16028,80 @@ pub(crate) mod tests {
         bank.process_transaction(&tx).unwrap();
     }
 
+    #[test]
+    fn test_failed_compute_request_instruction() {
+        solana_logger::setup();
+        let GenesisConfigInfo {
+            genesis_config,
+            mint_keypair,
+            ..
+        } = create_genesis_config_with_leader(
+            1_000_000_000_000_000,
+            &Pubkey::new_unique(),
+            bootstrap_validator_stake_lamports(),
+        );
+        let mut bank = Bank::new_for_tests(&genesis_config);
+
+        let payer0_keypair = Keypair::new();
+        let payer1_keypair = Keypair::new();
+        bank.transfer(10, &mint_keypair, &payer0_keypair.pubkey())
+            .unwrap();
+        bank.transfer(10, &mint_keypair, &payer1_keypair.pubkey())
+            .unwrap();
+
+        fn mock_ix_processor(
+            _first_instruction_account: usize,
+            invoke_context: &mut InvokeContext,
+        ) -> std::result::Result<(), InstructionError> {
+            let compute_budget = invoke_context.get_compute_budget();
+            assert_eq!(
+                *compute_budget,
+                ComputeBudget {
+                    max_units: 1,
+                    heap_size: Some(48 * 1024),
+                    ..ComputeBudget::default()
+                }
+            );
+            Ok(())
+        }
+        let program_id = solana_sdk::pubkey::new_rand();
+        bank.add_builtin("mock_program", &program_id, mock_ix_processor);
+
+        // This message will not be executed because the compute budget is request is invalid
```

Comment:
Are there other types of execution failures (from within `execute_loaded_transaction()` we should be testing?
I'm just wondering if the `signature_count` increment should actually be wrapped in a `if execution_result.was_executed()`

Reply:
That's a good idea, what do you think about moving it down to here: https://github.com/solana-labs/solana/blob/1e08178aa96a3a8d0d5649c23352f599bfeff41b/runtime/src/bank.rs#L4271

Reply:
Somewhat nevermind... it looks like the only way to get NotExecuted from within `execute_loaded_transaction()` is via account lifetime mismanagement.
However, is it maybe still worth wrapping in that `if` ^ for future-proofing?

Reply:
> what do you think about moving it down to here:

Yes, perfect